internal/proxy/directors.go

```diff
@@ -177,6 +178,64 @@ func (h DirectorSetHeaders) Director(request *http.Request) error {
 	return nil
 }
 
+type HTTPHeader struct {
+	Name  string
+	Value string
+}
+type HTTPHeaders []HTTPHeader
+type NetHeaders struct {
+	IPNet   net.IPNet
+	Headers HTTPHeaders
+}
+type DirectorSetHeadersByIP []NetHeaders
+
+func NewDirectorSetHeadersByIP(m map[string]HTTPHeaders) (DirectorSetHeadersByIP, error) {
+	res := make(DirectorSetHeadersByIP, 0, len(m))
+	for k, v := range m {
+		_, subnet, err := net.ParseCIDR(k)
+		if err != nil {
+			return nil, fmt.Errorf("can't parse CIDR: %v %w", k, err)
+		}
+
+		res = append(res, NetHeaders{
+			IPNet:   *subnet,
+			Headers: v,
+		})
+	}
+	return res, nil
```

Comment:
What if sort headers by IPNet, Header?

and while sort by IPNet - put outer networks before inner?

for example:
10.0.0.0/8, 10.0.0.0/24, 10.0.1.0/24, 10.2.0.0/24.


In current code order of these networks will random and rules with override headers will have different results between starts.

In suggested order - smaller network will always override bigger network rule.